sparse/unit_test/Test_Sparse_sptrsv.hpp

```diff
@@ -1151,207 +616,539 @@ void run_test_sptrsv_streams(int test_algo, int nstreams) {
+          khLd.get_sptrsv_handle()->print_algorithm();
+        } else {
+          std::cout << "Supernode Tri Solve SUCCESS" << std::endl;
+          khLd.get_sptrsv_handle()->print_algorithm();
+        }
+        EXPECT_TRUE(sum == scalar_t(X.extent(0)));
+
+        khLd.destroy_sptrsv_handle();
+        khUd.destroy_sptrsv_handle();
+      }
+#endif
+    }
+  }
 
-      KokkosSparse::spmv("N", ONE, triMtx, known_lhs, ZERO, rhs_v[i]);
-      Kokkos::fence();
+  static void run_test_sptrsv_streams(int test_algo, int nstreams) {
+    // Workaround for OpenMP: skip tests if concurrency < nstreams because of
+    // not enough resource to partition
+    bool run_streams_test = true;
+#ifdef KOKKOS_ENABLE_OPENMP
+    if (std::is_same<typename device::execution_space, Kokkos::OpenMP>::value) {
+      int exec_concurrency = execution_space().concurrency();
+      if (exec_concurrency < nstreams) {
+        run_streams_test = false;
+        std::cout << "  Skip stream test: concurrency = " << exec_concurrency
+                  << std::endl;
+      }
+    }
+#endif
+    if (!run_streams_test) return;
 
-      // Create handle
-      kh_v[i]           = KernelHandle();
-      bool is_lower_tri = true;
-      if (test_algo == 0)
-        kh_v[i].create_sptrsv_handle(SPTRSVAlgorithm::SEQLVLSCHD_RP, nrows,
-                                     is_lower_tri);
-      else if (test_algo == 1)
-        kh_v[i].create_sptrsv_handle(SPTRSVAlgorithm::SEQLVLSCHD_TP1, nrows,
-                                     is_lower_tri);
-      else
-        kh_v[i].create_sptrsv_handle(SPTRSVAlgorithm::SPTRSV_CUSPARSE, nrows,
-                                     is_lower_tri);
-
-      kh_ptr_v[i] = &kh_v[i];
-
-      // Symbolic phase
-      sptrsv_symbolic(kh_ptr_v[i], row_map_v[i], entries_v[i], values_v[i]);
-      Kokkos::fence();
-    }  // Done handle creation and sptrsv_symbolic on all streams
-
-    // Solve phase
-    sptrsv_solve_streams(instances, kh_ptr_v, row_map_v, entries_v, values_v,
-                         rhs_v, lhs_v);
-
-    for (int i = 0; i < nstreams; i++) instances[i].fence();
-
-    // Checking
-    for (int i = 0; i < nstreams; i++) {
-      scalar_t sum = 0.0;
-      Kokkos::parallel_reduce(
-          Kokkos::RangePolicy<typename device::execution_space>(
-              0, lhs_v[i].extent(0)),
-          ReductionCheck<ValuesType, scalar_t, lno_t>(lhs_v[i]), sum);
-      if (sum != lhs_v[i].extent(0)) {
-        std::cout << "Lower Tri Solve FAILURE on stream " << i << std::endl;
-        kh_v[i].get_sptrsv_handle()->print_algorithm();
+    scalar_t ZERO = scalar_t(0);
+    scalar_t ONE  = scalar_t(1);
+
+    const size_type nrows = 5;
+    const size_type nnz   = 10;
+
+    std::vector<execution_space> instances;
+    if (nstreams == 1)
+      instances = Kokkos::Experimental::partition_space(execution_space(), 1);
+    else if (nstreams == 2)
+      instances =
+          Kokkos::Experimental::partition_space(execution_space(), 1, 1);
```

Comment:
While you're refactoring this file, mind switching to this approach instead of the case-by-case for nstreams? This is really how it should be done in the GS test too.
```
    std::vector<int> weights(nstreams);
    std::fill(weights.begin(), weights.end(), 1);
    auto instances = Kokkos::Experimental::partition_space(execution_space(), weights);
```

Reply:
Actually I think that you can just use the constructor that take an initialization value to be more compact:
```
    std::vector<int> weights(nstreams, 1);
    auto instances = Kokkos::Experimental::partition_space(execution_space(), weights);
```

Reply:
Yes this is better 🤦

Reply:
Yep, we have optimized the crap out of that one line now hahaha

Reply:
Done. And I made the same change in the gauss seidel test.